File: server/php/src/Application/DTOs/AdditionalPaymentActions/RequestDto.php
<?php

namespace OnlinePayments\ExampleApp\Application\DTOs\AdditionalPaymentActions;

use OnlinePayments\ExampleApp\Application\Domain\Enums\Common\Currency;

final class RequestDto
{
    public int $amount;

    public ?Currency $currency = null;

    public ?bool $isFinal = null;

    public string $id;

    public function __construct(string $id)
    {
        $this->id = $id;
    }
}

File: server/ruby/app/business/dtos/additional_payment_actions/request_dto.rb
module Business
  module Dtos
    module AdditionalPaymentActions
      class RequestDto
        attr_accessor :amount, :currency, :is_final, :id

        def initialize(id:, amount: nil, currency: nil, is_final: nil)
          @id = id
          @amount = amount
          @currency = currency
          @is_final = is_final
        end
        def to_h
          {
            id: id,
            amount: amount,
            currency: currency,
            is_final: is_final
          }.compact
        end
      end
    end
  end
end